### DISCUSSION TITLE
Internal ESP32 temperature value is Fahrenheit but displays unit °C

### DISCUSSION THREAD

**Discussion by sternegugger** [arendst/Tasmota#14150] [Q&A] [answered]
I am using ESP32-WROOM-32 with tasmota32 version 9.5.0
and like to measure the room temperature.

The Web interface shows:
"ESP32 Temperature 70.6 °C"
![image](https://user-images.githubusercontent.com/24937182/147109919-93f9272d-f02b-4702-ace6-9b4613a2b513.png)
But my room temperature is 21.4 °C.
Calculating the displayed value 70.6 from Fahrenheit to Celsius results 21.4 °C, which is exactly my room temperature.
So I guess the internal temperature of my ESP32 is measured in Fahrenheit, but displayed with unit "°C".

In the documentation I found something like "CPU temperature".
I know the CPU temperature could be much higher than the room temperature.
But the ESP32 temperature is stable, independant of the CPU runnung time or CPU load.

Is this a bug or dependand of the used chip?
It would be nice to have this fixed. That would save me one extra temperature sensor.
![20211222_153751](https://user-images.githubusercontent.com/24937182/147112001-5a968aff-a159-4102-be1f-2226f0a49974.jpg)

**Comment by Jason2866** [comment, marked as answer]
It is not the room temperature. Depending on chip rev.? the shown temperature is garbage.

**Reply by sfromis** [reply]
Even if the sensor produced useful results, it would not be ambient temperature. I've seen lots of ESP32 temperature values neither making sense as Celsius nor Fahrenheit. As I see that nonsensical number as a nuisance, I'm normally disabling the ESP32 sensor function of Tasmota using `SetSensor127 0` (unless I want to have the Hall Effect sensor). 

Right now, on one board I'm seeing values of "48.9°C", which would be 9.9 C if read as F. I'm not sitting in such a cold room. While it in principle could make sense as a CPU temperature in C, fingering the chip (not encapsulated under a metal shield) reveals that it is not that hot either.

**Comment by bobybobybob** [comment]
The internal temperature values look quite good compared to external sensors nearby with a "constant" deviation of about 40-42 points. 

![ghobo](https://user-images.githubusercontent.com/23042717/216831606-083dc30d-4b87-4539-969a-46b02d20e32a.gif)

**Reply by sfromis** [reply]
In other words, it looks horrible. Anyway, it may not be consistent across ESP32 devices.

**Reply by scargill** [reply]
Hi SFROMIS - I'm writing about some NOUS products right now... they have various Tasmota-based controllers - and all show the ESP32 temperature - I wonder if you know, one shows "Temperature" - the other shows "ESP Temperature"... erm, how do you control that wording - any idea? I know how to turn that off the the webUI and how to change from C to F thanks to comments above and the Tasmota docs - but why would one product Tasmota webUI include the word ESP32 and the other not?

**Reply by arendst** [reply]
Because one product apparently has an additional external temperature sensor, reporting as `Temperature` and the other uses the (not recommended) ESP32 internal temperature sensor, reporting as `ESP32 Temperature`.

**Reply by scargill** [reply]
Thanks for the quick response - so on the subject of "Temperature" as against "ESP Temperature" - so I've just been in and had a look at the nous B3T - I can't see anything in their template for configuration - to indicate any kind of sensor other than the power monitoring chipBL0939 RX... the rest is just normal relayand button stuff. 2 relays, 2 buttons and a switch as well as ledlink and serial tx... anyway I'll  ask them - as I can't think of any other way to get a sensor in there. Can't see anything for the BL0939 about exposing temperature.

![image](https://github.com/user-attachments/assets/2495f287-cac0-4343-9b26-b43ccf40f4df)

**Reply by scargill** [reply]
AH, it seems the BLO939 has a temperature sensor... right.. https://evelta.com/content/datasheets/359-BL0939-SOP16L.pdf
